### PR TITLE
Nicely format json if possible

### PR DESCRIPTION
**Pull request by Commenter123**
Nicely format json responses from routes, if `new DataResponse($myArray)` is used :)

### CODE REVIEW

**Comment by LukasReschke** [comment]
Please add unit tests.

**Comment by Commenter123** [comment]
Why does this need unit tests?

**Comment by LukasReschke** [comment]
> Why does this need unit tests?

Because all changes to ownCloud require and this will also break the existing tests :wink:

**Comment by rullzer** [comment]
This will prettify all json responses. I don't think this is something we want. It has to be made optional

But why not just pipe the output trough: `python -m json.tool`? Or use the devtools of your browsers that also format received json just fine.

**Comment by PVince81** [comment]
Note that since the Sabre upgrade, the XML output of Sabre is also prettified. Not sure if it has an option, it might have become the default.

It doesn't hurt to have pretty JSON, so might be fine ?

I'm ok with this :+1:

**Comment by DeepDiver1975** [comment]
> Not sure if it has an option, it might have become the default.

AFAIK there used to be an option in the past ..... maybe we lost that option during the last upgrade ...

**Comment by MorrisJobke** [comment]
> AFAIK there used to be an option in the past

There is still an option - @nickvergessen introduced it.

**Comment by LukasReschke** [comment]
> I'm ok with this :+1:

Note that this still has failing unit tests.

**Comment by nickvergessen** [comment]
> There is still an option - @nickvergessen introduced it.

Only for the console I did, not for controllers.

Btw this might double+more the traffic, but if we think pretty of something that is not seen by users is important, fine by me.

File: lib/public/appframework/http/jsonresponse.php [hunk 1]
```diff
@@ -65,7 +65,14 @@ public function __construct($data=array(), $statusCode=Http::STATUS_OK) {
 	 * @throws \Exception If data could not get encoded
 	 */
 	public function render() {
-		$response = json_encode($this->data, JSON_HEX_TAG);
+		$options = JSON_HEX_TAG;
+		if (defined('JSON_UNESCAPED_SLASHES')) {
+			$options |= JSON_UNESCAPED_SLASHES;
```
Comment (nickvergessen) [inline lib/public/appframework/http/jsonresponse.php]: this is a behaviour change?

**Comment by nickvergessen** [comment]
Feel free to reopen this, when you want to fix the mentioned issues.

**Pull request closed without merging**